CharacterBuilder/src/app/api.service.ts

```diff
@@ -0,0 +1,20 @@
+import {firstValueFrom, map} from 'rxjs';
+import {ImageData} from './build/build.component';
+import {HttpClient} from '@angular/common/http';
+import {inject, Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  private readonly httpClient = inject(HttpClient);
+
+  async createImage(imageData: ImageData) {
+    return await firstValueFrom(this.httpClient.post('http://localhost:5110/build-image-url', imageData));
+  }
+
+  getRandomImage() : ImageData  {
```

Comment:
This return datatype is wrong. Like above, you should use `firstValueFrom`. The result would then be `Promise<something>`. You can find an example in https://github.com/rstropek/htl-2024-25-4AHIF/blob/0968ade4c631d34d13d0ca565738a13b85f9489d/homeworks/CharacterBuilder/ui/src/app/character-image.service.ts#L33